### PR TITLE
UP-4801: Autoformat Java Code to Google Code Style

### PR DESCRIPTION
**Pull request by ChristianMurphy**
##### Checklist

-   [x] the [individual contributor license agreement] is signed
-   [x] commit message follows [commit guidelines]

##### Description of change

[UP-4801](https://issues.jasig.org/browse/UP-4801)

:warning: This rewrites every Java file in the Project :warning: 

This uses [Google's Java Code Formatter](https://github.com/google/google-java-format) to automatically match uPortal's code with the Google code style.

``` sh
# Instructions used to generate this PR

# Download tool with dependencies
wget https://github.com/google/google-java-format/releases/download/google-java-format-1.2/google-java-format-1.2-all-deps.jar

# Run tool across codebase
java -jar ./google-java-format-1.2-all-deps.jar -replace **/*.java
```

[individual contributor license agreement]: https://github.com/Jasig/uPortal/blob/master/CONTRIBUTING.md#individual-contributor-license-agreement
[commit guidelines]: https://github.com/Jasig/uPortal/blob/master/CONTRIBUTING.md#commit

### CODE REVIEW

**Comment by ChristianMurphy** [comment]
> I'm *not looking forward* to the merge conflicts I assume this is going to create upon trying to merge versions of uPortal after this commit into UW-Madison's uPortal fork.

Since this is generated through an automated process. I would recommend 

merge all code ***before*** this commit as usual. Then for this commit
``` sh
git merge --strategy-option ours
```
to preserve all the changes to the fork, then run
``` sh
java -jar ./google-java-format-1.2-all-deps.jar -replace **/*.java
```
to re-apply the style to the forked files.

> Tradeoffs in deferring this change to the moment when pieces of the monolithic-heap-of-code are broken out into smaller GitHub repositories? Since that's going to require forks to reconsider the errors of their forking ways anyway?

I don't have a strong opinion on when it should be done.
I do strongly believe it should be done, and with some leeway for regression testing before it is officially released.

**Comment by bjagg** [comment]
Holding off on merge until next major PR merge session. Great stuff!

**Comment by drewwills** [comment]
@apetro @ChristianMurphy @bjagg 

    Tradeoffs in deferring this change to the moment when pieces of the monolithic-heap-of-code are broken out into smaller GitHub repositories?

We have reached precisely this point... sort of.

Next step in this evolution away from monolithic is breaking things into smaller compilation units that build to jars (though mostly still one repo).  That will me a lot of file moving.  Those changes will trigger the pain as well.  We might as well make this change pretty soon.

**Comment by drewwills** [comment]
I wasn't able to get the formatter to work with that command;  but I did get this one to work...

```
java -jar ./google-java-format-1.2-all-deps.jar -replace `find . -name "*.java"`
```

**Comment by bjagg** [comment]
I want to deviate from Google's guide in the case of spacing. It should really be 4 spaces.

**Comment by drewwills** [comment]
@ChristianMurphy @apetro @bjagg @asgrant @vertein @cousquer _et al._,

I think we're ready for a version of this pull request to be applied.  Other PRs, for the most part, have been merged, and we're ready to concentrate on "decomposing" uPortal sources into more tightly scoped jars.

But I regret to admit I have one remaining reservation about this change.  I tried to "get over it" -- I wanted to be bigger than this -- but I deeply dislike the look of Java code indented with just 2 spaces.  :(  I prefer what we've used previously:  4 spaces.

I actually think formatting the entire codebase with indentation using 2 spaces will interfere with my ability to read & understand it quickly.  It's just not easy to see where blocks begin & end.  Maybe it's just that my eyes are getting old... I'm not sure.

So I did some research.  (Here's a blog post I found helpful:  http://blog.takipi.com/tabs-vs-spaces-how-they-write-java-in-google-twitter-mozilla-and-pied-piper/)

### Here are some observations

  - The Google Java Style Guide appears to be in the minority;  other style guides (incl. Twitter & Mozilla) seem either (1) to offer a choice between 4 or 2 spaces or (2) to mandate 4 spaces
  - The majority of projects on GitHUb uses spaces (not tabs), and the majority of those use 4 of them
  - Google's other Java style guide -- the AOSP Coding Style (Android Java) -- mandates 4 spaces, not 2
  - The tool @ChristianMurphy used to prepare this PR -- the `google-java-format` tool -- has an option for 4 spaces instead of 2

### Here's what I propose

  - Use the Google tool with the 4 space option to format all the uPortal 5 code (master)
  - Check it in
  - Update the checkstyle & linting configs to match

I am happy to do all of that, if it helps.

**Comment by jgribonvald** [comment]
+1 for 4 spaces in java code please !  The 2 places are good on xml files.

**Comment by ChristianMurphy** [comment]
My concern would be with deviating from the styleguide, however since this is a part of the [android google java styleguide](https://source.android.com/source/code-style.html#use-spaces-for-indentation), this particular change *could* be okay. +0
If 4 spaces for indentation is the final decision, I would recommend that the style is applied not just to Java, but to JavaScript, Cascading Style Sheets, Less, and XML.

**Comment by bjagg** [comment]
My experience leads me to prefer +4 spaces for code and +2 spaces for data files for practical reasons. Specifically, Java and JavaScript blocks vary from method to method, and strong indentation provides clarity . OTOH XML (and pure HTML) have lengthy blocks of data that are more uniform. With +4 indentation, gains in clarity of relative location in a block are lost to high block depth causing wrapping or text extending past screen size.

CSS and Less are shallow and short enough that either indentation is fine.

While pure HTML (a'la templating) is closer to XML, practical HTML/JSP files may be better indented at +4. This is due to all the embedded JavaScript.

To summarize my view, I strongly prefer Java/JavaScript to be +4 indented, XML to be +2 indented and comfortable with either indentation for other files.

**Comment by ChristianMurphy** [comment]
I want to avoid this conversation getting off the rails. 🚋🛤 

The reason why an industry standard code style has been adopted is to avoid personal opinions causing deviance from the code style.
If there is a style guide which you believe would be better suited for the community, you are welcome to bring that into the discussion.

The Google code style is somewhat uniquely suited for uPortal, in that it offers [one guide with recommendations for CSS, HTML, Java, JavaScript, JSON, and XML (among others)](https://github.com/google/styleguide). Meaning one standard can be applied to a vast majority of the code in uPortal.

**Comment by bjagg** [comment]
Looks like there is only `-a` option for 4-space indentation: Nothing more refined to specify .java files only. 

Let's use that option.

**Comment by apetro** [comment]
I'm [`-0.9`](https://www.apache.org/foundation/voting.html#expressing-votes-1-0-1-and-fractions) on four spaces rather than two. The first rule of code style club is, [don't talk about code style club](http://apetro.ghost.io/citizens-for-a-better-uportal-tomorrow-tomorrow/#adoptgooglestyle).

**Comment by drewwills** [comment]
This work has been updated, committed, and pushed per discussion here.  (Merged out-of-band.)
